File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/people/PeoplePaths.java
package dev.reuss.tmdb.domain.people;

import dev.reuss.tmdb.value.id.PersonId;

final class PeoplePaths {

    private static final String PERSON = "/person";

    private PeoplePaths() {
    }

    static String details(PersonId personId) {
        return person(personId);
    }

    static String changes(PersonId personId) {
        return person(personId) + "/changes";
    }

    static String combinedCredits(PersonId personId) {
        return person(personId) + "/combined_credits";
    }

    static String externalIds(PersonId personId) {
        return person(personId) + "/external_ids";
    }

    static String images(PersonId personId) {
        return person(personId) + "/images";
    }

    static String movieCredits(PersonId personId) {
        return person(personId) + "/movie_credits";
    }

    static String tvCredits(PersonId personId) {
        return person(personId) + "/tv_credits";
    }

    static String translations(PersonId personId) {
        return person(personId) + "/translations";
    }

    static String latest() {
        return PERSON + "/latest";
    }

    static String popular() {
        return PERSON + "/popular";
    }

    private static String person(PersonId personId) {
        return PERSON + "/" + personId.asString();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/people/PeoplePaths.kt
package dev.reuss.tmdb.domain.people

import dev.reuss.tmdb.core.path.tmdbPath
import dev.reuss.tmdb.value.id.PersonId

internal object PeoplePaths {
    fun details(personId: PersonId): String = tmdbPath("person", personId)

    fun changes(personId: PersonId): String = tmdbPath("person", personId, "changes")

    fun combinedCredits(personId: PersonId): String = tmdbPath("person", personId, "combined_credits")

    fun externalIds(personId: PersonId): String = tmdbPath("person", personId, "external_ids")

    fun images(personId: PersonId): String = tmdbPath("person", personId, "images")

    fun movieCredits(personId: PersonId): String = tmdbPath("person", personId, "movie_credits")

    fun tvCredits(personId: PersonId): String = tmdbPath("person", personId, "tv_credits")

    fun translations(personId: PersonId): String = tmdbPath("person", personId, "translations")

    fun latest(): String = tmdbPath("person", "latest")

    fun popular(): String = tmdbPath("person", "popular")
}
